### PR TITLE
Make DepthwiseConv2D non-clusterable

### PR DESCRIPTION
**Pull request by arovir01**
This PR removes `tf.keras.layers.DepthwiseConv2D` from the `ClusteringRegistry`, as our experimental results suggest that depthwise convolution layers are not suited for clustering.

### CODE REVIEW

**Comment by arovir01** [comment]
@Ruomei

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper_test.py [hunk 1]
```diff
@@ -23,16 +23,13 @@
 from tensorflow_model_optimization.python.core.clustering.keras import cluster
 from tensorflow_model_optimization.python.core.clustering.keras import cluster_wrapper
 from tensorflow_model_optimization.python.core.clustering.keras import clusterable_layer
-from tensorflow_model_optimization.python.core.clustering.keras import clustering_registry
```
Comment (akarmi) [inline tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper_test.py:26]: Changes in this file seem unrelated to the topic. If it is a clean-up, consider moving them to a separate PR.

Comment (arovir01) [reply tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper_test.py:26]: It is indeed a leftover from a cleanup. Will move it to a separate change.

**Comment by Ruomei** [comment]
Since this is already unit tested, please let me know whether it works for mobilenet tests. Also, in the model-optimization repo, try "grep" DepthwiseConv2D or lower cases to see whether there are any other places that need to be cleaned up. For instance, in RFC, there is this "except DepthwiseConv2D layers"? do we want to keep it?

**Comment by akarmi** [comment]
> 
> 
> Since this is already unit tested, please let me know whether it works for mobilenet tests. Also, in the model-optimization repo, try "grep" DepthwiseConv2D or lower cases to see whether there are any other places that need to be cleaned up. For instance, in RFC, there is this "except DepthwiseConv2D layers"? do we want to keep it?

I would argue that we should keep the mentioning of DepthwiseConv2D in the RFC, as this makes it clear that not all layers with potentially clusterable parameters (weights) are actually clustered. Even more, this PR arguably makes the description of clustering configuraiton in that table in RFC more accurate, as it enables appying cluster_weights() to the whole model, and not per-layer, to reproduce the results in those particular rows.

**Comment by akarmi** [comment]
@alanchiao, please have a quick look. We decided to disable clustering DepthwiseConv2D layers by default as we have not seen any cases when it works well so far.

**Comment by alanchiao** [comment]
Seems fine to me. Once weight clustering is launched, these types of changes will be harder to make when considering backward compatibility.

While I wasn't there for the specific CL discussion (and don't know if how intentional it was), pruning also disables DepthwiseConv2D by default.
